protocols/dcutr/src/behaviour_impl.rs

```diff
@@ -246,32 +246,23 @@ impl NetworkBehaviour for Behaviour {
         local_addr: &Multiaddr,
         remote_addr: &Multiaddr,
     ) -> Result<THandler<Self>, ConnectionDenied> {
-        match self
-            .outgoing_direct_connection_attempts
```

Comment:
The reason we check this first is to ensure it is the relayed connection that _we_ (i.e. the dcutr behaviour) created.

Reply:
But `outgoing_direct_connection_attempts` gets an entry only after the relay is established and the `OutboundConnectNegotiated` event is completed. Or is my understanding incorrect?

Reply:
You are right, I misremembered !

Good catch! :)

Mind sending a separate PR just for this fix? Then we can focus on holepunching in here and land this earlier.